File: platforms/kotlin/examples/scene/material/emissive_texture.kt
import org.fragmentcolor.*

val renderer = Renderer()
val glow_bytes = listOf(255.0f, 0.0f, 0.0f, 255.0f, 255.0f, 0.0f, 0.0f, 255.0f, 255.0f, 0.0f, 0.0f, 255.0f, 255.0f, 0.0f, 0.0f, 255.0f, .0f)
val glow = renderer.createTexture(TextureInputMobile.Bytes(glow_bytes.let { ba -> ByteArray(ba.size) { i -> ba[i].toInt().and(0xFF).toByte() } }), null)
val mat = Material.pbr().emissive(listOf(0.8f, 0.0f, 0.0f)).emissiveTexture(glow)

File: platforms/swift/examples/scene/material/emissive_texture.swift
import FragmentColor

let renderer = Renderer()
let glow_bytes = [
    255, 0, 0, 255,
    255,   0, 0, 255,
    255,   0, 0, 255,
    255,   0, 0, 255,
]
let glow = try await renderer.createTexture((glow_bytes, [2, 2]))
let mat = try Material.pbr().emissive([0.8, 0.0, 0.0]).emissiveTexture(glow)